src/Microsoft.TemplateEngine.Edge/Settings/Scanner.cs

```diff
@@ -243,31 +260,40 @@ private bool ScanForComponents(IMountPoint mountPoint, string scanLocation, bool
 
                     if (typeList.Count > 0)
                     {
+                        // TODO: figure out what to do with probing path registration when components are not found.
+                        // They need to be registered for dependent assemblies, not just when an assembly can be loaded.
+                        // We'll need to figure out how to know when that is.
                         _environmentSettings.SettingsLoader.Components.RegisterMany(typeList);
                         _environmentSettings.SettingsLoader.AddProbingPath(Path.GetDirectoryName(asm.Key));
                         anythingFound = true;
                     }
                 }
                 catch
                 {
+                    // exceptions here are ok, due to dependency errors, etc.
                 }
             }
 
-            if (!anythingFound && !isInOriginalInstallLocation)
+            if (!anythingFound && isCopiedIntoContentDirectory)
             {
                 try
                 {
+                    // The source was copied to content and then scanned for components.
+                    // Nothing was found, and this is a copy that now has no use, so delete it.
+                    // Note: no mount pount was created for this copy, so no need to release it.
```

Comment:
nit: typo - should be "point"

Reply:
Heh, this has become one of my standard typos lately.